modules/dhcp/dhcp_api.rb

```diff
@@ -1,31 +1,16 @@
 class Proxy::DhcpApi < ::Sinatra::Base
+  extend Proxy::DHCP::DependencyInjection::Injectors
+
   helpers ::Proxy::Helpers
   authorize_with_trusted_hosts
   authorize_with_ssl_client
   use Rack::MethodOverride
 
+  inject_attr :dhcp_provider, :server
+
   before do
     begin
-      raise "Smart Proxy is not configured to support DHCP" unless Proxy::DhcpPlugin.settings.enabled
-      case Proxy::DhcpPlugin.settings.dhcp_vendor.downcase
-      when "isc"
-        require 'dhcp/providers/server/isc'
-        unless Proxy::DhcpPlugin.settings.dhcp_config && Proxy::DhcpPlugin.settings.dhcp_leases \
-          && File.exist?(Proxy::DhcpPlugin.settings.dhcp_config) && File.exist?(Proxy::DhcpPlugin.settings.dhcp_leases)
-          log_halt 400, "Unable to find the DHCP configuration or lease files"
-        end
-        @server = Proxy::DHCP::ISC.instance_with_default_parameters
-      when "native_ms"
-        require 'dhcp/providers/server/native_ms'
-        @server = Proxy::DHCP::NativeMS.instance_with_default_parameters
-      when "virsh"
-        require 'dhcp/providers/server/virsh'
-        @server = Proxy::DHCP::Virsh.instance_with_default_parameters
-      else
-        log_halt 400, "Unrecognized or missing DHCP vendor type: #{Proxy::DhcpPlugin.settings.dhcp_vendor.nil? ? "MISSING" : Proxy::DhcpPlugin.settings.dhcp_vendor}"
-      end
-
-      @server.loadSubnets
+      server.loadSubnets
```

Comment:
Would you mind doing some renaming of these methods?  I think now is going to be our only ever opportunity before plugins begin to be written against these camel-case method names.